scripts/tests.js

```diff
@@ -51,6 +41,9 @@ function createIntegrationMock() {
 
 createIntegrationMock();
 
+// HACK: allow us to change the `startDir()` during tests
```

Comment:
Aew, I actually forgot to remove this. In fact, I have to see this with @kulshekhar as this was to keep using `starDir()` but with a better detection on whether we are running our tests or not (the purpose of this special env var). Tho I also commented out the `startDir()`. I must chat more about this with @kulshekhar. The thing is that I reviewed totally the way `tsconfig` file is resolved, so there is no need to startDir() anymore. I also kinda isolated the tests so that they can't pickup the default `tsconfig.json` in the root of the project outside of the tests by renaming the `/tsconfig.sjon`.

For the `tsconfig` resolution (`jestConfig` is full as all missing props replaced by defaults/correct ones by jest upstream):

- jest config **has** `tsConfigFile`
    - `tsConfigFile` is asolute ➡️ use it directly
    - `tsConfigFile` has `<rootDir>` ➡️ replace `rootDir` and use the result
    - `tsConfigFile` is relative, use `path.resolve(jestConfig.rootDir, tsConfigFile)` _I wanted to use jestCOnfig.cwd but in multi-projects jest config it resolves to the path of the root jestConfig and not of each sub-project jest config, so `.` will refer to rootDir and not the path of jest config as it would break in multi-proects configs_
- jest config **does not have** `tsConfigFile`
    - starting from `jestConfig.rootDir`, try to find a `tsconfig.json`, is each parent dir. (talked with @kulshekhar about this: now it'll throw an error if it can't find a tsconfig file)

Reply:
Alright, so we're throwing this out?

Reply:
FYI the `startDir()` was returning `.` if not running tests, else `../../` or something like that